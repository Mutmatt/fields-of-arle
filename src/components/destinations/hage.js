import React from 'react'
import PropTypes from 'prop-types'

const isField = s => s.match(/^(flax-|grain-)/)

class Hage extends React.Component {
  constructor(props) {
    super()
  }

  render() {
    const { inventory, handleSellAtDestination } = this.props
    const fields = inventory.filter(isField)
    return (
      <div>
        <b>Sellers at Hage will buy:</b>
        <ul>
          <li>
            {fields.length !== 0 &&
              fields.map(field => (
                <button
                  key={field}
                  disabled={this.props.conversionInputs[0]}
                  type="button"
                  onClick={handleSellAtDestination(0, field)}
                >
                  {field}
                </button>
              ))}
            {fields.length === 0 && (
              <button disabled type="button">
                Field
              </button>
            )}
            for 2 food
          </li>
        </ul>
      </div>
    )
  }
}

Hage.propTypes = {
  conversionInputs: PropTypes.array.isRequired,
  inventory: PropTypes.array.isRequired,
  handleSellAtDestination: PropTypes.func.isRequired,
}

export default Hage
