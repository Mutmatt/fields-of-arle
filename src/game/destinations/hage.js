import { compose } from 'redux'
import { applyToCurrentPlayer } from '../common/player'

export const size = 1

export default conversionInputs => applyToCurrentPlayer(compose())
