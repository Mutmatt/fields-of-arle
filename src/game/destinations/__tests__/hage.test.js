import hage, { size } from '../hage'

const G = {
  players: {
    0: {
      land: [
        [
          { type: 'flax', contents: [] },
          { type: 'flax', contents: [] },
          { type: 'flax', contents: [] },
        ],
        [
          { type: 'flax', contents: [] },
          { type: 'flax', contents: [] },
          { type: 'flax', contents: [] },
        ],
        [
          { type: 'stall', contents: ['horse', 'horse'] },
          { type: 'empty', contents: [] },
          { type: 'flax', contents: [] },
        ],
      ],
      goods: {
        food: 0,
      },
    },
  },
}

const ctx = {
  currentPlayer: 0,
}

describe('hage load', () => {
  it('has a size', () => {
    expect(size).toEqual(1)
  })

  it('noop', () => {
    expect(hage([null])({ G, ctx })).toEqual({ G, ctx })
  })

  it('removes the field', () => {
    const player = hage([['flax-2-0']])({ G, ctx }).G.players['0']
    expect(player.land).toMatchSnapshot()
  })

  it('generates 1 food', () => {
    const player = hage([['flax-2-0']])({ G, ctx }).G.players['0']
    expect(player.goods.food).toBe(1)
  })
})
