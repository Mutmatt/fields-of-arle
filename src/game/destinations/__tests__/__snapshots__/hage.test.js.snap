// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hage load removes the field 1`] = `
Array [
  Array [
    Object {
      "contents": Array [],
      "type": "flax",
    },
    Object {
      "contents": Array [],
      "type": "flax",
    },
    Object {
      "contents": Array [],
      "type": "empty",
    },
  ],
  Array [
    Object {
      "contents": Array [],
      "type": "flax",
    },
    Object {
      "contents": Array [],
      "type": "flax",
    },
    Object {
      "contents": Array [],
      "type": "flax",
    },
  ],
  Array [
    Object {
      "contents": Array [
        "horse",
        "horse",
      ],
      "type": "stall",
    },
    Object {
      "contents": Array [],
      "type": "empty",
    },
    Object {
      "contents": Array [],
      "type": "flax",
    },
  ],
]
`;
